storage/cache/map_subtree_cache.go

```diff
@@ -61,7 +61,7 @@ func populateMapSubtreeNodes(treeID int64, hasher hashers.MapHasher) storage.Pop
 				return nil, nil
 			},
 			func(depth int, index *big.Int, h []byte) error {
-				if depth == 0 {
+				if depth == 0 && len(st.Prefix) > 0 {
```

Comment:
The batch test runs with 32 * 64 leaf nodes, which I think should be enough to exercise this case.  

The tests that I've added are a nice verification that this is correct: they fail without this change and pass post change.